chia/wallet/wallet_state_manager.py

```diff
@@ -529,13 +528,37 @@ async def synced(self):
             return True
         return False
 
-    def set_sync_mode(self, mode: bool, sync_height: uint32 = uint32(0)):
-        """
-        Sets the sync mode. This changes the behavior of the wallet node.
-        """
-        self.sync_mode = mode
-        self.sync_target = sync_height
-        self.state_changed("sync_changed")
+    @property
+    def sync_mode(self) -> bool:
+        return self._sync_target is not None
+
+    @property
+    def sync_target(self) -> Optional[uint32]:
+        return self._sync_target
+
+    @asynccontextmanager
+    async def set_sync_mode(self, target_height: uint32) -> AsyncIterator[None]:
+        self.log.debug(f"set_sync_mode enter {await self.blockchain.get_finished_sync_up_to()}-{target_height}")
+        async with self.lock:
+            start_time = time.time()
+            start_height = await self.blockchain.get_finished_sync_up_to()
+            self._sync_target = target_height
```

Comment:
previously we would not set `sync_mode = True` if `start_height - current_height <= self.LONG_SYNC_THRESHOLD`
. It seems this logic is gone now, and it's not obvious that's for the better, Is it deliberate?

Reply:
But only in trusted case.. in untrusted case we still set it even if we are not far behind. In case `synced_peers` is empty here https://github.com/Chia-Network/chia-blockchain/blob/bdbfc810a4ebedee506605dfa313043c5b11690b/chia/wallet/wallet_node.py#L1147